acceptance/status_server_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+// +build acceptance
+
+package acceptance
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/acceptance/localcluster"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+type details struct {
+	NodeID proto.NodeID `json:"nodeID"`
+}
+
+// get performs an HTTPS GET to the specified path for a specific node.
+func get(t *testing.T, client *http.Client, node *localcluster.Container, path string) []byte {
+	url := fmt.Sprintf("https://%s%s", node.Addr(""), path)
+	resp, err := client.Get(url)
+	if err != nil {
+		t.Fatalf("could not GET %s - %s", url, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("could not GET %s - statuscode: %d", url, resp.StatusCode)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("could not open body for %s - %s", url, err)
+	}
+	log.Infof("OK response from %s", url)
+	return body
+}
+
+func checkNode(t *testing.T, client *http.Client, node *localcluster.Container, nodeID, otherNodeID, expectedNodeID string) {
+	body := get(t, client, node, "/_status/details/"+otherNodeID)
+	var detail details
+	if err := json.Unmarshal(body, &detail); err != nil {
+		t.Fatalf("unable to parse details - %s", err)
```

Comment:
meta comment: using helper functions like this makes it exceedingly difficult to figure out what's going on when things fail because you don't get a stack trace.